monitor/main.go

```diff
@@ -461,12 +461,23 @@ func blockProcessor(ctx context.Context, headers <-chan *types.Header, cc *clien
 		transactionCtx := context.Background()
 		for _, txHash := range header.Transactions {
 			receipt, err := cc.Eth.TransactionReceipt(transactionCtx, txHash)
+			
 			if err != nil {
 				return err
 			}
 
 			txLogger := getTxLogger(logger, receipt, header)
-			logTransaction(txLogger)
+			
+			tx, isPending, err := cc.Eth.TransactionByHash(ctx, txHash)
+			
+			if err != nil {
+				return err
+			}
+
+			if !isPending {
```

Comment:
Is it possible for something to be pending here? As we're analysing a block already.

Reply:
No, it can't be pending here